ml-agents/mlagents/trainers/sac/trainer.py

```diff
@@ -47,75 +47,44 @@ def __init__(
         """
         super().__init__(
-            brain_name, trainer_parameters, training, run_id, reward_buff_cap
+            brain_name, trainer_settings, training, run_id, reward_buff_cap
         )
-        self.param_keys = [
-            "batch_size",
-            "buffer_size",
-            "buffer_init_steps",
-            "hidden_units",
-            "learning_rate",
-            "init_entcoef",
-            "max_steps",
-            "normalize",
-            "num_layers",
-            "time_horizon",
-            "steps_per_update",
-            "sequence_length",
-            "summary_freq",
-            "tau",
-            "use_recurrent",
-            "memory_size",
-            "output_path",
-            "reward_signals",
-        ]
-
-        self._check_param_keys()
+
         self.load = load
         self.seed = seed
         self.policy: NNPolicy = None  # type: ignore
         self.optimizer: SACOptimizer = None  # type: ignore
-
+        self.hyperparameters: SACSettings = cast(
+            SACSettings, trainer_settings.hyperparameters
+        )
         self.step = 0
 
         # Don't count buffer_init_steps in steps_per_update ratio, but also don't divide-by-0
-        self.update_steps = max(1, self.trainer_parameters["buffer_init_steps"])
-        self.reward_signal_update_steps = max(
-            1, self.trainer_parameters["buffer_init_steps"]
-        )
+        self.update_steps = max(1, self.hyperparameters.buffer_init_steps)
+        self.reward_signal_update_steps = max(1, self.hyperparameters.buffer_init_steps)
 
-        self.steps_per_update = (
-            trainer_parameters["steps_per_update"]
-            if "steps_per_update" in trainer_parameters
-            else DEFAULT_STEPS_PER_UPDATE
-        )
+        self.steps_per_update = self.hyperparameters.steps_per_update
         self.reward_signal_steps_per_update = (
-            trainer_parameters["reward_signals"]["reward_signal_steps_per_update"]
-            if "reward_signal_steps_per_update" in trainer_parameters["reward_signals"]
-            else self.steps_per_update
+            self.hyperparameters.reward_signal_steps_per_update
         )
 
-        self.checkpoint_replay_buffer = (
-            trainer_parameters["save_replay_buffer"]
-            if "save_replay_buffer" in trainer_parameters
-            else False
-        )
+        self.checkpoint_replay_buffer = self.hyperparameters.save_replay_buffer
 
     def _check_param_keys(self):
         super()._check_param_keys()
         # Check that batch size is greater than sequence length. Else, throw
         # an exception.
         if (
-            self.trainer_parameters["sequence_length"]
-            > self.trainer_parameters["batch_size"]
-            and self.trainer_parameters["use_recurrent"]
+            self.trainer_settings["sequence_length"]
+            > self.trainer_settings["batch_size"]
+            and self.trainer_settings["use_recurrent"]
```

Comment:
```suggestion
            self.trainer_settings.sequence_length
            > self.trainer_settings.batch_size
            and self.trainer_settings.use_recurrent
```

Reply:
Ah, this method isn't actually used anymore, the check is done in `settings.py` in `_check_batch_size_seq_length`. I've removed them from PPO and SAC.